src/streams/EditStreamCard.js

```diff
@@ -75,10 +86,18 @@ export default class EditStreamCard extends PureComponent<Props, State> {
         <ZulipButton
           style={styles.marginTop}
           text={isNewStream ? 'Create' : 'Update'}
-          disabled={name.length === 0}
+          disabled={
+            name.length === 0
+            || isMatch(initialValues, { name, description, invite_only: isPrivate })
```

Comment:
This looks like a lot of logic put in a property value assignment. Would be better as a flag near the `isInvalidStreamName` with a good name like `canUpdateStream` or any other appropriate and helpful name.

Reply:
`isMatch` looks like overcomplicating code that can be implemented with few `===`s.
While 'lodash' is popular and we do not shy away from using it, I definitely had to go an read the documentation to be able to understand if the code is correct.

This code looks simpler and more exact as:
````js
initialValues.name === name 
&& initialValues. description === description  
&& initialValues. invite_only === isPrivate
```

Reply:
Makes sense. I pushed a new version. Thank you!